src/workloads/query/TenMDocCollection_IntId.yml

```diff
@@ -1,109 +1,224 @@
+      Key: ActorTemplateConfig
+      Parameters:
+        MaxPhases: *MaxPhases
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {id: {$eq: {^RandomInt: {min: 0, max: 9999999}}}}
+- TemplateName: IDHACKIn
+  Config:
+    LoadConfig:
+      Path: TenMDocCollection_Base.tmpl
+      Key: ActorTemplateConfig
+      Parameters:
+        MaxPhases: *MaxPhases
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {_id: {$in: {^FixedGeneratedValue: {fromGenerator: {^Array: {distinct: true, number: 100000, of: {^RandomInt: {min: 0, max: 9999999}}}}}}}}
+- TemplateName: IntFieldIn
+  Config:
+    LoadConfig:
+      Path: TenMDocCollection_Base.tmpl
+      Key: ActorTemplateConfig
+      Parameters:
+        MaxPhases: *MaxPhases
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {id: {$in: {^FixedGeneratedValue: {fromGenerator: {^Array: {distinct: true, number: 100000, of: {^RandomInt: {min: 0, max: 9999999}}}}}}}}
+- TemplateName: IDHACKOr
+  Config:
+    LoadConfig:
+      Path: TenMDocCollection_Base.tmpl
+      Key: ActorTemplateConfig
+      Parameters:
+        MaxPhases: *MaxPhases
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {$or: {^FixedGeneratedValue: {fromGenerator: {^Array: {distinct: true, number: 100000, of: {_id: {^RandomInt: {min: 0, max: 9999999}}}}}}}}
+- TemplateName: IntFieldOr
+  Config:
+    LoadConfig:
+      Path: TenMDocCollection_Base.tmpl
+      Key: ActorTemplateConfig
+      Parameters:
+        MaxPhases: *MaxPhases
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {$or: {^FixedGeneratedValue: {fromGenerator: {^Array: {distinct: true, number: 100000, of: {id: {^RandomInt: {min: 0, max: 9999999}}}}}}}}
 
+Actors:
+- LoadConfig:
+    Path: TenMDocCollection_Base.tmpl
+    Key: ClearCollectionActor
+    Parameters:
+      MaxPhases: *MaxPhases
+- LoadConfig:
+    Path: TenMDocCollection_Base.tmpl
+    Key: PrepopulateActor
+    Parameters:
+      Document:
+        # The multiplier is multiplied by the ActorID, and added to the value.
+        # Since this is the second actor, the ActorIDs increment for each thread starting at 2.
+        # e.g -2000000 + 2 * 1000000 = 0, -1999999 + 2 * 1000000 = 1, etc.
+        # Once complete, the ids the collection will have be the range 0-9999999.
+        _id: {^Inc: {start: -2000000, multiplier: 1000000}}
+        id: {^Inc: {start: -2000000, multiplier: 1000000}}
+      Indexes:
+        - keys: {id: 1}
+      MaxPhases: *MaxPhases
+- LoadConfig:
+    Path: TenMDocCollection_Base.tmpl
+    Key: QuiesceActor
+    Parameters:
+      MaxPhases: *MaxPhases
 - ActorFromTemplate:
-    TemplateName: IDHACK
+    TemplateName: IDHACKEq
     TemplateParameters:
-      Name: IDHack32Threads
-      Threads: 32
+      Name: IDHACKEq032Threads
```

Comment:
Did you mean to add the 0 here? Also I think the "Eq" is not necessarily needed since only equalities are eligible for IDHACK. I worry it may be more confusing, since the name sort of implies there might be an IDHACKLt or IDHACKIn

Reply:
Oh, I see that this is indeed a topic of confusion :) Try loading the dataset (it doesn't go away after your test run), and running some explains with these queries. I believe that you should only see the IDHACK plan stage in the explain output for the existing workloads - I don't know of another way to generate that plan stage aside from exact equality on _id.

Reply:
Oh.

The `0` in `032` is intentional, so that when sorting on the actor name, the 32-thread runs will show up first. The `0`s in front of the other thread counts is unintentional.

Reply:
Is there any point in retaining the `id` field, then? Shouldn't we be just testing on `_id` anyway?

Reply:
Actually, I take that back, the eq test still needs to happen, right?

Reply:
We want the `id` field so that we can compare a single point lookup between IDHACK and "regular" index scans. The non-_id point lookup is probably the cheapest query you could imagine otherwise. These other, more expensive queries are more interesting as "other representative queries" which will presumably make us feel better that our "worst case" performance is indeed worse than "average case" or something like that.

Actually after typing that out, there probably is a cheaper one: An equality query with a projection on only that field, so that we would not need to have a FETCH in the query plan - all the data comes from that one location in the index. Could we add that also?